ocaml/jane/doc/proposals/kind-inference.md

```diff
@@ -197,22 +200,23 @@ k = κ ∈ Γ
 ---------------------------------------- K_LAYOUT
 Γ ⊢ const_layout ↠ const_layout; ⟪⊤_Ξ⟫
 
-Γ ⊢ τ : κ
------------------------------------------- K_OF
-Γ ⊢ kind_of_ τ ↠ layout_of τ; ⟪⊥_Ξ with τ⟫
-
 Γ ⊢ jkind ↠ κ
 ------------------------------------------------- K_MOD
 Γ ⊢ jkind mod modes ↠ lay(κ); ⟪Ξ(κ) ⊓ ⨅ extract(Ξ, modes)⟫
-  (* CR reisenberg: This rule isn't right if κ already has [with] constraints; probably
-     should just forbid that syntactically. Not right: we can't meet out
-     the with-added modes because we don't really know what they are.
-     So we'd have to retain the [mod] clause just like we retain the
-     [with] clause, which is terrible. *)
```

Comment:
I have the same question, and I don't understand how it was resolved by this change. What does happen if κ already has [with] constraints?

Reply:
The user-written syntax indeed prevents `with` constraints from ending up in the \kappa here. But the system does not track this. I have split out \chi from \kappa, where \chi does not have `with` constraints. This is a nice improvement. I also may have noticed a few other small bugs that I fixed while taking a pass through.